cmd/krel/cmd/promote-images.go

```diff
@@ -187,11 +182,19 @@ func runPromote(opts *promoteOptions) error {
 			return errors.Wrap(err, "while reading the current promoter image list")
 		}
 	}
+	if len(oldlist) == 0 {
+		return errors.Wrap(err, "no image list found")
+	} 
+	manifest, err := reg.ParseManifest(oldlist)
```

Comment:
I think we have to populate the options and grow the manifest like done there:
https://github.com/kubernetes-sigs/k8s-container-image-promoter/blob/4b2f14a1108a9cec649af5785ae4fbafac36a3e5/cmd/cip-mm/main.go#L76-L90

Reply:
I couldn't find a `type GrowManifestOptions struct` in sigs.k8s.io/k8s-container-image-promoter/lib/dockerregistry: v1.0.7 

Should I try to find the version that does have this and put one in?

Reply:
Looks like we're already at version 2.x.x, but there seems to be many changes as well. Can you try to integrate the master branch?